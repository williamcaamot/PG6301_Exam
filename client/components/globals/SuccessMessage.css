.successMessage {
    background-color: #ddffdd;
    border-left: 6px solid #4CAF50;
    padding: 12px;
    color: #155724;
    border-radius: 4px;
    font-family: Arial, sans-serif;
    font-size: 16px;
    line-height: 24px;
    box-shadow: 0 2px 4px rgba(0,0,0,0.2);
}