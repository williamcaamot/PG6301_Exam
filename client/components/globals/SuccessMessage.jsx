import React from "react";

function SuccessMessage(props) {
  if (props.message) {
    return (
      <>
        <div
          style={{ width: "100%", paddingTop: "10px", paddingBottom: "10px" }}
        >
          <div className={"successMessage"}>
            <p>{props.message}</p>
          </div>
        </div>
      </>
    );
  }
}

export default SuccessMessage;
