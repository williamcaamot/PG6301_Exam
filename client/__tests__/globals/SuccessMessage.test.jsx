import renderer, { act } from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import SuccessMessage from "../../components/globals/SuccessMessage.jsx";

describe("error message", () => {
  it("show a success message!", () => {
    const message = "This is a success message!";

    const component = renderer.create(
      <MemoryRouter>
        <SuccessMessage message={message} />
      </MemoryRouter>,
    );
    expect(component).toMatchSnapshot();
  });
});
