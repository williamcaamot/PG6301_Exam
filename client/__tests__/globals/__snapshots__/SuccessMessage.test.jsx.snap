// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error message show a success message! 1`] = `
<div
  style={
    {
      "paddingBottom": "10px",
      "paddingTop": "10px",
      "width": "100%",
    }
  }
>
  <div
    className="successMessage"
  >
    <p>
      This is a success message!
    </p>
  </div>
</div>
`;
